src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSource.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Getter
+@Setter
+public class BasicGremlinSource implements GremlinSource {
+
+    private String id;
+    private String label;
+    private Map<String, Object> properties;
+
+    @Setter(AccessLevel.PRIVATE)
+    private GremlinScript script;
+
+    public BasicGremlinSource() {
+        this.id = null;
+        this.label = null;
+        this.properties = new HashMap<>();
+        this.script = null;
+    }
+
+    public void setGremlinScriptStrategy(@NonNull GremlinScript script) {
+        this.setGremlinScriptStrategy(script);
+    }
+
+    public Object generateGremlinScript() {
+        return this.script.generateScript(this);
```

Comment:
Dunno what is this generated script, so it is impossible to narrow the type here?